packages/react/src/host/controller.tsx

```diff
@@ -23,9 +32,9 @@ export function createController(
       return value;
     },
     renderer: {
-      renderComponent,
-      renderText,
-      ...renderer,
+      renderComponent:
+        componentRenderer?.(defaultRenderComponent) ?? defaultRenderComponent,
+      renderText: textRenderer?.(defaultRenderText) ?? defaultRenderText,
```

Comment:
I was thinking more like this type signature:

```
interface RendererOptions {
  renderComponent?(details: Parameters<Renderer['renderComponent']>[0], renderDefault: () => ReactNode): ReactNode;
}
```

So a consumer could intentionally return `null` in some cases, and so they are always explicitly returning and not just falling through their function as a way of signalling to use the default. If you guys strongly prefer this approach, it works for me too, I think it's just a slightly more complicated signature since it means the consumer is dealing with some extra closures.

Reply:
Ah. If you look closer to the commit change, you will see we did exactly what you want but in slightly different way. The consumer absolutely can return null for some cases.